File: fm-domain/src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/UserInfoResponse.java
package io.zeitwert.fm.app.adapter.api.rest.dto;

import lombok.Builder;
import lombok.Data;

import java.util.List;

import io.zeitwert.dddrive.ddd.api.rest.dto.EnumeratedDto;

@Data
@Builder
public class UserInfoResponse {

	private Integer id;
	private String name;
	private String email;

	private EnumeratedDto role;
	private List<EnumeratedDto> tenants;

}

File: fm-domain/src/main/java/io/zeitwert/fm/app/adapter/api/rest/dto/UserInfoResponse.kt
package io.zeitwert.fm.app.adapter.api.rest.dto

import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.dto.EnumeratedDto
import io.zeitwert.dddrive.ddd.adapter.api.jsonapi.dto.TypedEnumeratedDto

data class UserInfoResponse(
	val id: Int,
	val name: String,
	val email: String,
	val role: EnumeratedDto,
	val tenants: List<TypedEnumeratedDto>,
)
